File: 2729. Check if The Number is Fascinating/Solution.py
class Solution:
    def isFascinating(self, n: int) -> bool:
        s = str(n) + str(2 * n) + str(3 * n)
        a = [c for c in s]
        if any([b == '0' for b in a]): return False
        return len(set(a)) == 9 and len(a) == 9

File: 2729. Check if The Number is Fascinating/Solution.java
class Solution {
    public boolean isFascinating(int n) {
        String s = Integer.toString(n) + Integer.toString(2 * n) + Integer.toString(3 * n);
        char[] c = s.toCharArray();
        Set<Character> seen = new HashSet<>();
        for (char ch : c) {
            seen.add(ch);
            if (ch == '0') return false;
        }
        return seen.size() == 9 && c.length == 9;
    }
}
